cli/file_fetcher.rs

```diff
@@ -145,6 +145,45 @@ pub fn get_source_from_bytes(
   Ok(source)
 }
 
+fn get_source_from_data_url(
+  specifier: &ModuleSpecifier,
+) -> Result<(String, MediaType, String), AnyError> {
+  let url = specifier.as_url();
+  if url.scheme() != "data" {
+    return Err(custom_error(
+      "BadScheme",
+      format!("Unexpected scheme of \"{}\"", url.scheme()),
+    ));
+  }
+  if url.query().is_some() {
+    return Err(custom_error(
+      "BadUrl",
+      "The data URL includes a query string which is not supported.",
+    ));
+  }
+  let path = url.path();
+  let mut parts = path.splitn(2, ',');
+  let media_type_part = parts.next().unwrap();
+  let data_part = if let Some(data) = parts.next() {
+    data
+  } else {
+    return Err(custom_error(
+      "BadUrl",
+      "The data URL is badly formed, missing a comma.",
+    ));
+  };
+  let (media_type, maybe_charset) =
+    map_content_type(specifier, Some(media_type_part.to_string()));
+  let is_base64 = media_type_part.rsplit(';').any(|p| p == "base64");
```

Comment:
Which is exactly what this does, it only checks the last segment after the semi-colon.  Treating it arbitrarily like it isn't part of the content type is really immaterial for the code logic.

Reply:
I wonder what's the behaviour on such a URL: `data:text/plain;base64;charset=UTF-8,SGVsbG8sIFdvcmxkIQ==`. 

From the spec this should not parse as base64, but given the logic below it would. The iterator returned by `.rsplit(';')` would contain `["charset=UTF-8", "base64"]`. As `any` of them, namely the last, is equal to `"base64"`, the data would be treated as such.

That's my understanding but I may have missed something. It's also not a big deal; the likelihood of such an edge case to ever cause any issue is zero.